powerscale/provider/network_setting_resource_test.go

```diff
@@ -0,0 +1,354 @@
+/*
+Copyright (c) 2023 Dell Inc., or its subsidiaries. All Rights Reserved.
+
+Licensed under the Mozilla Public License Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://mozilla.org/MPL/2.0/
+
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	powerscale "dell/powerscale-go-client"
+	"fmt"
+	"regexp"
+	"terraform-provider-powerscale/client"
+	"terraform-provider-powerscale/powerscale/helper"
+	"terraform-provider-powerscale/powerscale/models"
+	"testing"
+
+	. "github.com/bytedance/mockey"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/stretchr/testify/assert"
+)
+
+var networkSettingGetMocker *Mocker
+var networkSettingMocker *Mocker
+
+func TestAccNetworkSettingResourceImport(t *testing.T) {
+	var networkSettingResourceName = "powerscale_network_settings.test"
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: ProviderConfig + networkSettingBasicResourceConfig,
+			},
+			// Import testing
+			{
+				Config:            ProviderConfig + networkSettingBasicResourceConfig,
+				ResourceName:      networkSettingResourceName,
+				ImportState:       true,
+				ExpectError:       nil,
+				ImportStateVerify: true,
+				ImportStateCheck: func(s []*terraform.InstanceState) error {
+					assert.Equal(t, "groupnet0", s[0].Attributes["default_groupnet"])
+					return nil
+				},
+			},
+			// Update and Read testing
+			{
+				Config: ProviderConfig + networkSettingUpdateResourceConfig,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(networkSettingResourceName, "source_based_routing_enabled", "true"),
```

Comment:
Will this values change sometime? Is it better to check that values bool/number, etc?

Reply:
In the config of networkSettingUpdateResourceConfig, source_based_routing_enabled is setted to true;  So the value here should be true

Reply:
@baoy1 In the config of networkSettingUpdateResourceConfig, source_based_routing_enabled is setted to true; So the value here should be true

Reply:
NP. I thought the value will change sometimes.